examples/md-flexible/src/ParallelVtkWriter.cpp

```diff
@@ -139,9 +139,47 @@ void ParallelVtkWriter::recordParticleStates(size_t currentIteration,
 
   // print positions
   timestepFile << "        <DataArray Name=\"positions\" NumberOfComponents=\"3\" format=\"ascii\" type=\"Float32\">\n";
+  const auto boxMax = autoPasContainer.getBoxMax();
   for (auto particle = autoPasContainer.begin(autopas::IteratorBehavior::owned); particle.isValid(); ++particle) {
-    auto pos = particle->getR();
-    timestepFile << "        " << pos[0] << " " << pos[1] << " " << pos[2] << "\n";
+    // When we write to the file in ASCII, values are rounded to the precision of the filestream.
+    // Since a higher precision results in larger files because more characters are written,
+    // and mdflex is not intended as a perfectly precice tool for application scientists,
+    // we are fine with the rather low default precision.
+    // However, if a particle is very close to the domain border it can happen that the particle position is rounded
+    // exactly to the boundary position. This then causes problems when the checkpoint is loaded because boxMax is
+    // considered to be not part of the domain, hence such a particle would not be loaded and thus be lost.
+    // This function identifies these problematic values and raises the write precision just for this value high enough
+    // to be distinguishable from the boundary.
+    const auto writeWithDynamicPrecision = [&](double d, double border) {
+      const auto initialPrecision = timestepFile.precision();
+      // Simple and cheap check if we even need to do anything.
+      if (border - d < 0.1) {
```

Comment:
by default values are written with floating point precision of 6 digits to the vtk file. Thus, the "problematic delta" can be arbitrarily large and I wanted to ignore the more expensive logic further down.

Since the check is very arbitrary indeed we can also get rid of it because it has the potential to cause problems for domains that extend to ~1e6.

Reply:
Ah I didn't realise that this is only used with `boxMax`